src/testplane.ts

```diff
@@ -154,21 +154,19 @@ export class Testplane extends BaseTestplane {
 
     protected async _readFailed(): Promise<FailedListItem[]> {
         try {
-            const data = await fs.readFile(this._config.failedTestsPath, "utf8");
-            return JSON.parse(data);
+            const data = await fs.readJSON(this._config.failedTestsPath);
+            return data;
         } catch {
+            //No error because it may be convinient to always use --run-failed, even on the first run
+            logger.warn(`WARNING: Could not read ${this._config.failedTestsPath}`);
             return [];
         }
     }
 
     protected async _saveFailed(): Promise<void> {
         const dirname = path.dirname(this._config.failedTestsPath);
-        try {
-            await fs.access(dirname);
-        } catch {
-            await fs.mkdir(dirname, { recursive: true });
-        }
-        await fs.writeFile(this._config.failedTestsPath, JSON.stringify(this.failedList));
+        await fs.ensureDir(dirname);
+        await fs.writeJSON(this._config.failedTestsPath, this.failedList, { spaces: 4 });
```

Comment:
Just found out "fs-extra" has "outputJSON" method. It does the same thing as "writeJSON", but also creates the directory if it does not exist: https://github.com/jprichardson/node-fs-extra/blob/master/docs/outputJson.md

So in "_saveFailed" you could just call 
```ts
await fs.outputJSON(this._config.failedTestsPath, this.failedList, { spaces: 4 })
```
Without "path" library methods